sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -349,29 +349,38 @@ object CatalogTable {
 
 /**
  * This class of statistics is used in [[CatalogTable]] to interact with metastore.
+ * We prefer Spark statistics over Hive statistics if they both exist.
  * We define this new class instead of directly using [[Statistics]] here because there are no
- * concepts of attributes or broadcast hint in catalog.
+ * concepts of attributes or hints in catalog.
  */
 case class CatalogStatistics(
-    sizeInBytes: BigInt,
-    rowCount: Option[BigInt] = None,
-    colStats: Map[String, ColumnStat] = Map.empty) {
+    sparkStats: Option[ExternalStatistics] = None,
+    hiveStats: Option[ExternalStatistics] = None) {
```

Comment:
If we are designing the interface like this, we might need to refactor it again in the near future. Stats could be collected from Spark, imported from Hive, set by external users, or even from the data source API v2 (in the future).